File: src/components/old.ts
import { t } from "@rbxts/t";
import defineComponent from "../../src/defineComponent";

const definition = defineComponent({
	name: "Old",
	data: <
		{
			[key in Exclude<keyof Components, "Old">]?: Components[key]["data"];
		}
	>true,

	refValidator: t.any,
});

declare global {
	interface Components {
		/** The Old Component contains a Map that holds old values of other components */
		Old: typeof definition;
	}
}

export = definition;

File: src/components/old.lua
local TS = _G[script]
local t = TS.import(script, TS.getModule(script, "@rbxts", "t").lib.ts).t
local defineComponent = TS.import(script, script.Parent.Parent, "defineComponent").default
local definition = defineComponent({
	name = "Old",
	data = true,
	refValidator = t.any,
})
return definition
